File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        l=[]
        co_map={')':'(', ']':'[', '}':'{'}
        for b in s:
            if b in co_map.values():
                l.append(b)
            if b in co_map.keys():
                if len(l)==0 or (l.pop()!=co_map[b]):
                    return False
        return True if len(l)==0 else False
        

File: Data Structures & Algorithms/daily-temperatures/submission-3.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temps) {
        vector<int> result(temps.size(), 0);
        // strictly decreasing stack -> what we maintain 
        // in C++, a stl stack pop does not return the popped value, use top for that 
        stack<int> s;
        for (int i=0;i<temps.size();i++){
            // make sure the current element isnt larget than anything 
            while (s.size()>0 && temps[s.top()]<temps[i]){
                int top = s.top();
                result[top]=i-top;
                s.pop();
            }
            // now adding the current value maintains the strictly decreasing order
            s.push(i);
        }
        return result;
    }
};
